### PR TITLE
.Net: [Error handling] [Part10] Using HttpOperationException for Http operations

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
This PR is one of the follow-up PRs aimed at gradually replacing SK custom exceptions with SKException to provide a simple, consistent, and extensible exception model. You can find more details in the [ADR](https://github.com/microsoft/semantic-kernel/blob/main/docs/decisions/0004-error-handling.md) discussing error handling.

### Description
Translate HttpRequestException and ServiceException into HttpOperationException for all SK components that use HTTP, allowing SK consumer code to consistently manage HTTP-related exceptions.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs [hunk 1]
```diff
@@ -303,15 +300,13 @@ public async Task UpdateAsync(string indexName, PineconeDocument document, strin
...
         try
         {
-            response.EnsureSuccessStatusCode();
+            await this.ExecuteHttpRequestAsync(basePath, request, cancellationToken).ConfigureAwait(false);
         }
-        catch (HttpRequestException e)
+        catch (HttpOperationException e)
         {
-            this._logger.LogWarning("Vector update for Document {0} failed. Message: {1}", document.Id, e.Message);
+            this._logger.LogWarning("Vector update for Document {Id} failed. {Message}", document.Id, e.Message);
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs:309]: I think this is the place where exception handling and telemetry tasks overlap :) 
I noticed in some catch blocks we log as `Warning` and `Debug`. Since it's critical error, could you please update such places to log as `Error`, so telemetry will be consistent? Feel free to skip it and we can do that in scope of separate PR.

File: dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs [hunk 2]
```diff
@@ -330,15 +325,15 @@ public async Task UpdateAsync(string indexName, PineconeDocument document, strin
...
         try
         {
-            response.EnsureSuccessStatusCode();
+            (_, responseContent) = await this.ExecuteHttpRequestAsync(basePath, request, cancellationToken).ConfigureAwait(false);
         }
-        catch (HttpRequestException e)
+        catch (HttpOperationException e)
         {
-            this._logger.LogDebug("Index not found {0}", e.Message);
+            this._logger.LogDebug("Index not found {Message}", e.Message);
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs:336]: Same here, in the same class the log level is inconsistent. If you already updated string template in these places, maybe it's worth to update log level as well. It can be also done in separate PR.

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs]: Easy fix, done.

File: dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs (hunk 1 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs]: Easy fix, done.

**Review by lemillermicrosoft — APPROVED** [review]
